src/main/kotlin/org/kamiblue/client/gui/hudgui/elements/world/Radar.kt

```diff
@@ -0,0 +1,95 @@
+
+import net.minecraft.entity.EntityLivingBase
+import org.kamiblue.client.event.KamiEventBus.post
+import org.kamiblue.client.event.SafeClientEvent
+import org.kamiblue.client.event.events.RenderRadarEvent
+import org.kamiblue.client.gui.hudgui.HudElement
+import org.kamiblue.client.manager.managers.FriendManager
+import org.kamiblue.client.module.modules.client.GuiColors
+import org.kamiblue.client.setting.GuiConfig.setting
+import org.kamiblue.client.util.EntityUtils
+import org.kamiblue.client.util.EntityUtils.isNeutral
+import org.kamiblue.client.util.EntityUtils.isPassive
+import org.kamiblue.client.util.color.ColorHolder
+import org.kamiblue.client.util.graphics.RenderUtils2D.drawCircleFilled
+import org.kamiblue.client.util.graphics.RenderUtils2D.drawCircleOutline
+import org.kamiblue.client.util.graphics.VertexHelper
+import org.kamiblue.client.util.graphics.font.FontRenderAdapter
+import org.kamiblue.client.util.math.Vec2d
+import org.kamiblue.client.util.threads.runSafe
+import org.lwjgl.opengl.GL11.glRotatef
+import org.lwjgl.opengl.GL11.glTranslated
+import kotlin.math.abs
+import kotlin.math.min
+
+object Radar : HudElement(
+    name = "Radar",
+    category = Category.WORLD,
+    description = "Shows entities and new chunks"
+) {
+
+    private val radarScale by setting("Radar scale", 3f, 1f..10f, 0.1f)
+
+    /* Entity type settings */
+    private val players = setting("Players", true)
+    private val passive = setting("Passive Mobs", false)
+    private val neutral = setting("Neutral Mobs", true)
+    private val hostile = setting("Hostile Mobs", true)
+    private val invisible = setting("Invisible Entities", true)
+
+    override val hudWidth: Float = 130.0f
+    override val hudHeight: Float = 130.0f
+
+    private val radius get() = min(hudWidth, hudHeight) / 2
+
+    override fun renderHud(vertexHelper: VertexHelper) {
+        super.renderHud(vertexHelper)
+        runSafe {
+            drawBorder(vertexHelper)
+            post(RenderRadarEvent(vertexHelper, radius, radarScale)) //Let other modules display radar elements
+            drawEntities(vertexHelper)
+            drawLabels()
+        }
+    }
+
+    private fun SafeClientEvent.drawBorder(vertexHelper: VertexHelper) {
+        glTranslated(radius.toDouble(), radius.toDouble(), 0.0)
+        drawCircleFilled(vertexHelper, radius = radius.toDouble(), color = GuiColors.backGround)
+        drawCircleOutline(vertexHelper, radius = radius.toDouble(), lineWidth = 1.8f, color = primaryColor)
+        glRotatef(player.rotationYaw + 180, 0f, 0f, -1f)
+    }
+
+    private fun SafeClientEvent.drawEntities(vertexHelper: VertexHelper) {
+        drawCircleFilled(vertexHelper, radius = 1.0, color = primaryColor) //player marker
+
+        val playerTargets = arrayOf(players.value, true, true) //enable friends and sleeping
+        val mobTargets = arrayOf(true, passive.value, neutral.value, hostile.value) //enable mobs
```

Comment:
Spaces after `//` please, and begin the sentence with a capital letter preferably